_posts/2018-01-05-power-of-memory.md

```diff
@@ -0,0 +1,195 @@
+question:
+
+**What is the best memory one strategy against a given other memory one strategy?**
+
+<p align="center">
+  <img src="{{site.baseurl}}/assets/images/against_one_player.png" style='height: 20%; width: 30%; object-fit: contain'>
+</p>
+
+A match between two memory one players \\(p\\) and \\(q\\) can be modelled as a
+stochastic process, where the players move from state to  state. More specifically,
+it can be model by the use of a [Markov chain](https://en.wikipedia.org/wiki/Markov_chain),
+which is described by matrix \\(M\\).
+
+\\[M =
+\begin{bmatrix}
+    p_{1} q_{1} & p_{1} (- q_{1} + 1) & q_{1} (- p_{1} + 1) & (- p_{1} + 1) (- q_{1} + 1)
+    \\\
+    p_{2} q_{3} & p_{2} (- q_{3} + 1) & q_{3} (- p_{2} + 1) & (- p_{2} + 1) (- q_{3} + 1)
+    \\\
+    p_{3} q_{2} & p_{3} (- q_{2} + 1) & q_{2} (- p_{3} + 1) & (- p_{3} + 1) (- q_{2} + 1)
+    \\\
+    p_{4} q_{4} & p_{4} (- q_{4} + 1) & q_{4} (- p_{4} + 1) & (- p_{4} + 1) (- q_{4} + 1)
+    \\\
+\end{bmatrix}
+\\]
+
+Our players are assumed to move from each state until the system reaches a
+state steady. Thus in order to calculate the utility of our players we
+multiply the steady states of \\(M\\) with the payoffs of the game matrix.
+
+More specifically, the first result discussed here is that the utility of player
+\\(p\\) against an opponent \\(q\\), denoted as \\(u_q(p)\\), can be
+written in compact way:
+
+\\[u_q(p) = \frac{\frac{1}{2}p^TQ + c^Tp + a}
+            {\frac{1}{2}p^T\bar{Q} + \bar{c}^Tp + \bar{a}}\\]
+
+where \\(Q, \bar{Q}\\) are matrices of \\(4 \times 4\\), and \\(c, \bar{c}\\) are
+\\(4 \times 1\\) vectors defined with the transition probabilities of the
+opponent's transition probabilities \\(q_1, q_2, q_3, q_4\\).
+
+Thus in order to answer our question, `what is the best memory one strategy against
+a given other memory one strategy?`, we need to consider the following optimization
+problem:
+
+\\[ max_{q}: u_q(p) \\]
+\\[st: \\ p \in\mathbb{R}_{[0,1]}^{4}\\]
+
+Solving analytically a maximisation problem of \\(16\\) can be proven to be
+complicated. Thus in this blog post and as a second part of my work I considered
+a constrained version of the above problem. This was mainly done to get a better
+understanding of how to tackle the bigger problem.
+
+More specifically, the constrained problem explored here is that of a set of memory one
+strategies where the transition probabilities of each state are the same. These
+are called **purely random strategies** and can be denoted as \\(p = (p, p, p, p)\\).
+Now our optimisation problem can be re-written as:
+
+\\[ \max_p: u_q(p) = \frac{n_2p^2 + n_1p +n_0 } {d_1p + d_0}\\]
+\\[ p_1 = p_2 = p_3 = p_4 = p \\]
+\\[ \ 0 \leq p \leq 1 \\]
+
+where \\(n_2, n_1, n_0, d_1, d_0\\) are constants defined with the transition
+probabilities of the opponent's transition probabilities \\(q_1, q_2, q_3, q_4\\).
+
+Determining \\(p^*\\) for a given \\(q\\) becomes trivial. It can be proven that
```

Comment:
`becomes simpler` (very close with `trivial` but a bit too strong in this case: if it's trivial there's no real reason to even talk about it.)